File: backend/app/models/together.py
from __future__ import annotations

from pydantic import BaseModel, Field

from app.models.mood import MoodType, TimeSlot


class CreateRoomRequest(BaseModel):
    name: str = Field(default="Guest", max_length=24)
    favourite_ids: list[str] = Field(default_factory=list, max_length=100)


class JoinRoomRequest(BaseModel):
    name: str = Field(default="Guest", max_length=24)
    favourite_ids: list[str] = Field(default_factory=list, max_length=100)


class RoomPickRequest(BaseModel):
    mood: MoodType
    time_available: TimeSlot


class MemberView(BaseModel):
    """What everyone in the room may see about a member — a name and how much
    taste they brought, never the titles themselves."""

    id: str
    name: str
    favourite_count: int


class RoomState(BaseModel):
    code: str
    members: list[MemberView]
    combined_favourites: int
    expires_in: int  # seconds until the room lapses if idle


class JoinResult(BaseModel):
    member_id: str
    room: RoomState

File: frontend/src/api/together.ts
import api from './client';
import type { RecommendationResponse } from '../types/recommendation';
import type { MoodType, TimeSlot } from '../types/mood';

export interface MemberView {
  id: string;
  name: string;
  favourite_count: number;
}

export interface RoomState {
  code: string;
  members: MemberView[];
  combined_favourites: number;
  expires_in: number;
}

export interface JoinResult {
  member_id: string;
  room: RoomState;
}

export const createRoom = (name: string, favourite_ids: string[]) =>
  api.post<JoinResult>('/api/together/rooms', { name, favourite_ids });

export const joinRoom = (code: string, name: string, favourite_ids: string[]) =>
  api.post<JoinResult>(`/api/together/rooms/${code}/join`, { name, favourite_ids });

export const getRoom = (code: string) =>
  api.get<RoomState>(`/api/together/rooms/${code}`);

export const roomPick = (code: string, mood: MoodType, time_available: TimeSlot) =>
  api.post<RecommendationResponse>(`/api/together/rooms/${code}/pick`, { mood, time_available });
